cosmos/config.py

```diff
@@ -146,6 +147,7 @@ class ProjectConfig:
     seeds_path: Path | None = None
     snapshots_path: Path | None = None
     project_name: str
+    remote_cache_path: Path | None = None
```

Comment:
Previously, we attempted to have a single cache directory, at an Airflow deployment level:
<img width="1508" alt="Screenshot 2024-08-14 at 09 55 51" src="https://github.com/user-attachments/assets/2ba0d53f-9bf0-4deb-99e1-8d15e8daafa3">

https://astronomer.github.io/astronomer-cosmos/configuration/partial-parsing.html#caching

Does it make sense to do the same here?

Perhaps we can also have the cache subpath in the `ProjectConfig` If we want to give users the granularity to define a subpath per project - but in that case, I believe it should be `cache_subpath` - , so it can be used both remotely and locally as well.

Reply:
We still would have a single cache directory for the Airflow deployment. `ProjectConfig` is just a safe place I found it to retrieve and configure the `remote_cache_path` (we can find for a better place). 

Previously I was configuring it in the settings.py file itself https://github.com/astronomer/astronomer-cosmos/pull/1147/commits/ae0f45586912f1ccbfa7677c18720bdc781ea139#diff-2317d3342b8470278e95b3fcc555507acb062a595c518396ad5b94f1d3021e3eR41, but having it at the top level gave errors by the fsspec module complaining about it not being fork-safe.

We however are not exposing this param in the ProjectConfig but just using it for Cosmos internals. Can discuss more over our call.

Reply:
yes, we have a single remote_cache_dir as discussed. Removed from ProjectConfig and added a method to the `cache.py` module as suggested, thanks!